record/record.go

```diff
@@ -198,9 +231,43 @@ func (r *Reader) Next() (io.Reader, error) {
 		return nil, r.err
 	}
 	r.started = true
+	r.inRecovery = false
 	return singleReader{r, r.seq}, nil
 }
 
+// Attempt to recover from a corrupted record by continuing at the next block
+// boundary. Recover() clears internal error flags. Calling Recover() before
+// Next() raises an error, is itself an error. Recover() is only applicable for
+// data corruption issues. Calling recover if Next() returned io.EOF or
+// io.ErrUnexpectedEOF will result in an error.
+func (r *Reader) Recover() error {
+	switch r.err {
+	case nil:
+		return errors.New("leveldb/record no error condition exists to recover from")
+	case io.EOF, io.ErrUnexpectedEOF:
+		return errors.New("leveldb/record can't recover from io.EOF and/or io.UnexpectedEOF")
+	}
+
+	// After calling r.Recover(), we're effectively starting fresh, so we need to
+	// set r.inRecovery to true so r.Next() and r.nextChunk() proceed with the next block.
+	// We also set r.{i,j,n} = 0 and r.err = nil so we don't reparse the old block's data.
+	r.i, r.j, r.n = 0, 0, 0
+	r.err = nil
+	r.inRecovery = true
+
+	return nil
+}
+
+// If the underlying io.Reader implements io.ReadSeeker, return the offset at which the next
+// block will be read. This does not mean all data before the returned offset has been consumed
+// at the time CurrentOffset() was called.
+func (r *Reader) CurrentOffset() (int64, error) {
```

Comment:
It can, CurrentOffset was just more readable than r.rs.Seek(0, os.SEEK_CUR) imho.  If however it's not idiomatic Go, happy to replace it.